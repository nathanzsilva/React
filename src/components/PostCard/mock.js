export const postCardPropsMock = {
    title: 'title1',
    body: 'body1',
    id: 1,
    cover: 'img/img.png'
}