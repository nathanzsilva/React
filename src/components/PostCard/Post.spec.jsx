import { render, screen } from "@testing-library/react"
import { PostCard } from "."
import { postCardPropsMock } from "./mock"

const props = postCardPropsMock;

describe("<PostCard />", () => {
    test("should render PostCard correctly", () => {
        render(<PostCard {...props} />)

        expect.assertions(3);

        expect(screen.getByAltText(/title1/i))
            .toHaveAttribute('src', props.cover);

        expect(screen.getByRole('heading', { name: 'title1 1' })).toBeInTheDocument();

        expect(screen.getByText('body1')).toBeInTheDocument();
    })
    
    test("should match snapshot", () => {
        const { container } = render(<PostCard {...props} />)
        expect(container.firstChild).toMatchSnapshot();
    })
}) 