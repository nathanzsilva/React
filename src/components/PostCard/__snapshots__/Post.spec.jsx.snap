// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      title1
       
      1
    </h2>
    <p>
      body1
    </p>
  </div>
</div>
`;
